flocker/docs/_version.py

```diff
@@ -0,0 +1,64 @@
+# -*- test-case-name: flocker.docs.test.test_version -*-
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+import re
+
+_VERSION_RE = re.compile(
+    # The base version (in 'release')
+    r"(?P<release>[0-9]\.[0-9]+\.[0-9]+)"
+    # For a documentation release, the release number.
+    # (in 'doc-release').
+    r"(\+doc\.(?P<doc>[0-9]+))?"
+    # For development version, the number of commits since the last
+    # release and git hash
+    r"((?P<development>-[0-9]+-g[0-9a-f]+))?"
+    # Wether the tree is dirty.
+    r"((?P<dirty>-dirty))?"
+    # Always match the entire version string.
+    r"$"
+    )
+
+
+class UnparseableVersion(Exception):
+    """
+    A version was passed that was unable to be parsed.
+    """
+
+
+def parse_version(version):
```

Comment:
This seems like it would be nicer if, instead of a dictionary, there was a Version object, and characteristic did the comparison work.